**ДЗ семинар 7.   
Непараметрические тесты.**    
***1-4 задачи решать с помощью функций, 5ю вручную.    
Выбрать тест и проверить, есть ли различия между выборками***

*Задача 1:    
Даны **две независимые** выборки. Не соблюдается условие нормальности   
x1 380,420, 290   
y1 140,360,200,900   
Сделайте вывод по результатам, полученным с помощью функции*

Для данного условия задачи выбираем критерий Манна-Уитни.   
Гипотеза $\displaystyle H_0$: $µ_1 = µ_2$   
Гипотеза $\displaystyle H_0$: $µ_1 ≠ µ_2$  
где $µ_1$ и $µ_2$ - медианы перовй и второй выборок.    
Воспользуемся функцией Манна-Уитни (stats.Mannwhitneyu())

In [1]:
import numpy as np
import scipy.stats as stats

In [2]:
x1 = np.array([380, 420, 290])
y1 = np.array([140, 360, 200, 900])

In [3]:
stats.mannwhitneyu(x1, y1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

В результате вычисления функции получено значение p-value = 0,6285   
При условии, что уровень значимости α = 0,05, полученное значение p-value > α   
    
*Вывод: т.к. p-value > α, принимаем гипотезу $\displaystyle H_0$, т.е. статистичеси значемые различия между выборками x1 и y1  отсутствуют.*

*Задача 2:    
Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?   
1е измерение до приема препарата: 150, 160, 165, 145, 155   
2е измерение через 10 минут: 140, 155, 150, 130, 135   
3е измерение через 30 минут: 130, 130, 120, 130, 125*

Для рассчёта выбираем критерий Фридмана (stats.friedmanchisquare()), т.к. даны **множественные зависимые** выборки.   
А также уровень значимости α = 0,05

In [4]:
before = np.array([150, 160, 165, 145, 155])
test1 = np.array([140, 155, 150, 130, 135])
test2 = np.array([130, 130, 120, 130, 125])

In [5]:
stats.friedmanchisquare(before, test1, test2)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

В результате вычисления функции получено значение p-value = 0.0083   
При условии, что уровень значимости α = 0,05, полученное значение p-value < α   
    
*Вывод: т.к. p-value < α, принимаем гипотезу $\displaystyle H_1$, т.е. обнаружены статистичеси значемые различия между выборками.    
Препарат влияет на уровень давления пациентов.*

*Задача 3:    
Для условия предыдущей задачи 2, сравните 1-е и 2-е измерения, предполагая, что 3-го измерения через 30 минут не было.*

Имеем **две зависимые** выбоки:    
1-е измерение до приема препарата: 150, 160, 165, 145, 155   
2-е измерение через 10 минут: 140, 155, 150, 130, 135   

Для рассчёта выбираем критерий Уилкоксона (stats.wilcoxon(x, y))   
Также принимаем уровень значимости α = 0,05  

In [8]:
before = np.array([150, 160, 165, 145, 155])
after = np.array([140, 155, 150, 130, 135])

In [9]:
stats.wilcoxon(before, after)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

В результате вычисления функции получено значение p-value = 0.0625   
При условии, что уровень значимости α = 0,05, полученное значение p-value > α   
    
*Вывод: т.к. p-value > α, принимаем гипотезу $\displaystyle H_0$, т.е. статистичеси значемые различия между выборками не обнаружены.    
Препарат не оказал статистически значимого влияния на давление пациентов после повторного измерения давления через 10 минут после приёма препарата.*

*Задача 4: 
Даны 3 группы учеников плавания.   
В 1 группе время на дистанцию 50 м составляет: 56, 60, 62, 55, 71, 67, 59, 58, 64, 67   
Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53   
Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54   
Есть ли различия между группами?*

Даны **множественные независимые** выборки.   
Для рассчёта выбираем критерий Крускала–Уоллиса (stats.kruskal())
Также принимаем уровень значимости α = 0,05   

In [10]:
gr1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
gr2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
gr3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

In [11]:
stats.kruskal(gr1, gr2, gr3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

В результате вычисления функции получено значение p-value = 0.065   
При условии, что уровень значимости α = 0,05, полученное значение p-value > α   
    
*Вывод: т.к. p-value > α, принимаем гипотезу $\displaystyle H_0$, т.е. статистичеси значемые различия между выборками не обнаружены.    
Статистически значимых различий между результатами учеников по плаванию не обнаружено.*

*Задача 5: 
Заявляется, что партия изготавливается со средним арифметическим 2,5 см.    
Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения.    
Объем выборки 10, уровень статистической значимости 5%   
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34*

Гипотеза $\displaystyle H_0$: $µ = µ_0$, где $\mu$ и $\mu_0$ средние для выборки и генеральной совокупности, соответственно   
Гипотеза $\displaystyle H_0$: $µ ≠ µ_0$  
$µ$ = 2,5 см.   
n = 10   
α = 0,05

Выбираем тест по критерию Стьюдента, т.к. σ неизвестна. 

$t_н=\frac{\bar X-\mu}{\sigma_н/\sqrt n}$

Найдем среднее значение $\bar X$ для данной выборки: 


In [30]:
a = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
mu = 2.5
x = sum(a)/len(a)
x

2.5279999999999996

Найдём $\sigma_н = \sqrt{\frac {\sum_{i=1}^n * (x_i - \bar x)^2}{(n-1)}}$

In [28]:
def sigma_styudent(a, x):
    sigma = 0
    sum_1 = 0
    for i in range(len(a)):
        sum_1 += (a[i] - x)**2
    return (sum_1/(len(a)-1))**0.5

In [29]:
sigma_styudent(a, x)

0.1572542173961923

Рассчитаем $t_н=\frac{\bar X-\mu}{\sigma_н/\sqrt n}$

In [37]:
tn = (x - mu)/(sigma_styudent(a, x)/(len(a))**0.5)
print(f'Наблюдаемое значение кртерия Стьюдента {tn:.4f}')

Наблюдаемое значение кртерия Стьюдента 0.5631


Получим табличное значение критерия Стьюдента $t_t$ для $\alpha/2=0.05/2=0.025$ : 

In [36]:
tt = stats.t.ppf(0.975, len(a)-1)
print(f'Табличное значение кртерия Стьюдента {tt:.4f}')

Табличное значение кртерия Стьюдента 2.2622


В результате расчетов $t_н$ = 0.5631,  $t_t$ = 2.2622   
Таким образом $t_н$ < $t_t$

*Вывод: Принимаем гипотезу $\displaystyle H_0$ - среднее по выборке не имеет статистических отклонений от среднего по генеральной совокупности на уровне значимости 0,05.*